GUIDELINES.md

```diff
@@ -0,0 +1,79 @@
+# 📑 Engineering Guidelines
+
+## ✅ Testing
+
+**Don't optimise for coverage, optimise for well-designed tests.**
+
+Write positive and negative unit tests.
+
+- Write positive unit tests for things that the code should handle. Validate any states that change as a result of these tests.
+- Write negative unit tests for things that the code should not handle. It is helpful to follow up on the positive test (as an adjacent test) and make the change needed to make it pass.
+- Each code path should have its own unit test.
+
+Any addition or change to the code must be accompanied by relevant and comprehensive tests. Refactors should avoid simultaneous changes to tests.
+
+The test suite should run automatically for each change in the repository, and for pull requests, the tests must succeed before merging.
+
+Please consider writing [Foundry](https://github.com/foundry-rs/foundry)-based unit tests, property-based tests (i.e. fuzzing), and invariant tests for all contracts, if applicable.
+
+## 🪅 Code Style
+
+🐍Vyper code should be written in a consistent format that follows our [🐍Vyper Conventions](#vyper-conventions).
+
+Solidity test code should be written in a consistent format enforced by a linter and prettier that follows the official [Solidity Style Guide](https://docs.soliditylang.org/en/latest/style-guide.html). Also, we refer to Foundry's [best practices](https://book.getfoundry.sh/tutorials/best-practices?highlight=security#best-practices).
+
+The code should be simple and straightforward, with a focus on readability and comprehensibility. Consistency and predictability should be maintained throughout the code base. This is especially true for naming, which should be systematic, clear, and concise.
+
+Wherever possible, modularity, composability, and gas efficiency should be pursued, but not at the expense of security compromises.
+
+## 🛠 Pull Request (PR) Format
+
+Pull requests are _squash-merged_ in most cases to keep the `main` branch history clean. The title of the PR becomes the commit message, so it should be written in a consistent format:
+
+- Start with an emoji that well describes the PR.
```

Comment:
Consider adding a specific list of items here. For example:

- 📝 - documentation
- ♻️ - refactor
- 👷 - ci/cd
- 🐛 - bug
- ✨ - feature add
- ⚡️ - optimization

Reply:
noted and enhanced. That's the list:
  - Bug Fix 🐛
  - CI/CD 👷‍♂️
  - Documentation 📖
  - Events 🔊
  - Gas Optimisation ⚡️
  - New Feature 💥
  - Nit 🥢
  - Refactor/Cleanup ♻️
  - Security Fix 🔒